lib/llm/src/perf.rs

```diff
@@ -340,6 +348,61 @@ pub fn record_response_stream<R: Data + Clone>(
     record_stream(response_stream, mode)
 }
 
+/// Record a data stream by consuming it entirely and returning the recorded data.
+///
+/// This is a convenience function that operates only in "sink" mode - it consumes
+/// the entire stream and returns all the recorded responses. Unlike the other recording
+/// functions, this doesn't return a pass-through stream.
+///
+/// # Arguments
+/// * `data_stream` - The data stream to consume and record
+///
+/// # Returns
+/// A `RecordedStream` containing all the responses with timing information
+///
+/// # Example
+/// ```rust,ignore
+/// use dynamo_llm::perf::record_data_stream;
+///
+/// let stream = create_my_data_stream();
+/// let recorded = record_data_stream(stream).await;
+/// println!("Recorded {} responses", recorded.response_count());
+/// ```
+pub async fn record_data_stream<R: Data + Clone>(
+    mut data_stream: DataStream<R>,
+) -> RecordedStream<R> {
+    use futures::StreamExt;
+
+    let start_time = Instant::now();
+    let mut responses = Vec::new();
+    let mut sequence_number = 0;
+
+    // Consume the entire stream
+    while let Some(item) = data_stream.next().await {
+        responses.push(TimestampedResponse::new(item, sequence_number));
+        sequence_number += 1;
+    }
+
+    let end_time = Instant::now();
+
+    RecordedStream::new(responses, start_time, end_time)
+}
+
+/// Read an annotated stream from a file and collect all items
+pub fn read_annotated_stream_from_file<T: Serialize + DeserializeOwned>(
+    path: &str,
+) -> Result<DataStream<Annotated<T>>, anyhow::Error> {
+    // Read the entire file as a string
+    let data =
+        std::fs::read_to_string(path).map_err(|e| anyhow::anyhow!("Failed to read file: {}", e))?;
```

Comment:
Can you put `path` in the error? Otherwise we get the annoying UNIX "No such file or directory" but it doesn't say which one.

Reply:
done